index.js

```diff
@@ -69,7 +62,9 @@ function ensureAllCommentsPrinted(astComments) {
 }
 
 function formatWithCursor(text, opts, addAlignmentSize) {
-  if (opts.requirePragma && !hasPragma(text)) {
+  const selectedParser = parser.resolveParser(opts);
+  const hasPragma = !selectedParser.hasPragma || selectedParser.hasPragma(text);
+  if (opts.requirePragma && !hasPragma) {
     return { formatted: text };
```

Comment:
Should this return the current cursor position as well as the original text?

Reply:
I don't know, unrelated to this PR